File: php/rooms/RoomBuilder.php
<?php

namespace rooms;

use chest\ChestType;
use monsters\CreatureType;
use rooms\options\GameMasterable;
use rooms\options\RoomType;

interface RoomBuilder {
    public function createEmptyRoom(RoomType $type);
    public function addCreature(CreatureType $type);
    public function addChest(ChestType $type);

    public function reset();
    public function build(): GameMasterable;
}
?>

File: java/rooms/RoomBuilder.java
package rooms;

import chest.ChestType;
import monsters.CreatureType;
import rooms.options.GameMasterable;
import rooms.options.RoomType;

public interface RoomBuilder {
    void createEmptyRoom(RoomType type);
    void addCreature(CreatureType type);
    void addChest(ChestType type);

    void reset();
    GameMasterable build();
}
